diff --git a/compiler/rustc_llvm/build.rs b/compiler/rustc_llvm/build.rs
--- a/compiler/rustc_llvm/build.rs
+++ b/compiler/rustc_llvm/build.rs
@@ -283,6 +283,9 @@ fn detect_llvm_link() -> (&'static str, &'static str) {
         if name == "LLVMLineEditor" {
             continue;
         }
+        if name == "Polly" || name == "PollyISL" {
+            continue;
+        }
 
         let kind = if name.starts_with("LLVM") { llvm_kind } else { "dylib" };
         println!("cargo:rustc-link-lib={kind}={name}");
